preload

```diff
@@ -110,130 +85,161 @@ if [ ! -f "$listfile" ] || [ -z "$(cat "$listfile")" ];then
+  APPS="$(grep -v '/' <<<"$vfiles" | list_intersect "$(list_apps cpu_installable)")"
   
   #get list of folders - excluding apps - and hide the hidden folder.
-  DIRS="$(echo "$vfiles" | grep '/' | tr -d '/' | grep -vx "hidden")"
-  
-  #If updates available, show special Updates category
-  if "${DIRECTORY}/updater" get-status &>/dev/null;then
-    DIRS="Updates
-$DIRS"
-  fi
+  DIRS="$(grep '/' <<<"$vfiles" | tr -d '/' | grep -vFx "hidden")"
   
   #shuffle the list if enabled
   if [ "$(cat "${DIRECTORY}/data/settings/Shuffle App list")" == 'Yes' ];then
     APPS="$(echo "$APPS" | shuf)"
     DIRS="$(echo "$DIRS" | shuf)"
   fi
   
-  #remove apps that are not compatible with OS architecture
-  APPS="$(echo "$APPS" | list_intersect "$(list_apps cpu_installable)")"
-  
   if [ "$format" == yad ];then
     IFS=$'\n'
-    LIST=''
-    for i in $DIRS
+    
+    #initial value of listfile: if within a prefix, start with a Back button
+    if [ ! -z "$prefix" ];then
+      echo "${DIRECTORY}/icons/none-1.png
+${DIRECTORY}/icons/back.png
+Back
+$(dirname "$prefix" | sed 's+^\.$++g')/
+Return to the previous location" | tee "$listfile"
+    else
+      echo -n '' > "$listfile"
+    fi
+    
+    for dir in $DIRS
     do
-      if [ -f "${DIRECTORY}/icons/categories/${i}.png" ];then
-        diricon="${DIRECTORY}/icons/categories/${i}.png"
+      if [ -f "${DIRECTORY}/icons/categories/${dir}.png" ];then
+        diricon="${DIRECTORY}/icons/categories/${dir}.png"
       else
         diricon="${DIRECTORY}/icons/categories/default.png"
       fi
       
-      LIST="${LIST}$diricon
-${DIRECTORY}/icons/none-1.png
-$i
-$i/
+      add="${DIRECTORY}/icons/none-1.png
+$diricon
+$dir
+$prefix$dir/
 App folder
 "
-    done #finished preloading categories
+      echo -n "$add"
+    done | sed 's/&/&amp;/g' | tee -a "$listfile"
+    #finished preloading categories
     
-    #set to false to disable genapplist-yad program - use the original bash version
-    if true && [ -f "${DIRECTORY}/etc/genapplist-yad" ];then
-      LIST="$LIST$(DIRECTORY="$DIRECTORY" APPS="$APPS" "${DIRECTORY}/etc/genapplist-yad")
-"
-    else
-      for i in $APPS
-      do
-        status1="$(cat "${DIRECTORY}/data/status/${i}" 2>/dev/null || echo "none")"
-        status2="$(echo "$status1" | sed 's/none/uninstalled/g')"
-        
-        LIST="$LIST$(echo "${DIRECTORY}/icons/$status1.png")
-${DIRECTORY}/apps/${i}/icon-24.png
-$i
-$i
-"\("$status2"\)" $(head -n1 "${DIRECTORY}/apps/${i}/description" || echo "Description unavailable")
-"
-      done
-    fi #finished preloading apps
+    #preload apps
+    for app in $APPS
+    do
+      #get installation status of app
+      status="$(cat "${DIRECTORY}/data/status/${app}" 2>/dev/null)"
+      
+      #determine status icon for app
+      if [ -z "$status" ];then
+        #app has never been installed, so no icon
+        add="${DIRECTORY}/icons/none-1.png"$'\n'
```

Comment:
I honestly think this looks kinda jarring (the no icon) next to uninstalled icons. Maybe we could add a bag (or whatever the pi-apps logo is supposed to be) with nothing in it (no ❌  and no ✔️ ) for apps that haven't been installed or uninstalled

Reply:
Hmm. Okay

Reply:
Basically, I considered the perspective of an average new user. According to our analytics, around three apps are installed for every Pi-Apps download. Not very many.

When a new user first runs Pi-Apps, I realized that it would be quite distracting to see a long list of red ❌'s. Also it would be hard to spot which apps have been installed, and subsequently uninstalled.
Removing the default status icon from the list, I believe, improves the first-impressions as it cuts down on visual noise.

@theofficialgman, You say it's "jarring." Is that simply because it's different than what you're used to?
Maybe I could do an empty bag by default, but that would increase visual noise, especially for first launch.

Reply:
Part of me wants to remove the status icons altogether.
![2022-03-09-140458_1920x1080_scrot](https://user-images.githubusercontent.com/54716352/157524802-d592fc91-82e5-4f57-adaf-2e0b96b3b56e.png)
Though that may frustrate our power-users who use Pi-Apps frequently and depend on the status icons.

Reply:
can we change the color of the text for each app? maybe just do that to represent install status (green for installed, etc)

Reply:
> can we change the color of the text for each app? maybe just do that to represent install status (green for installed, etc)

Great idea!

Reply:
> can we change the color of the text for each app?

I recently made a yad-based accounting software and found that you *can* change a row's color.
How does this look?
![2022-03-09-145540_1920x1080_scrot](https://user-images.githubusercontent.com/54716352/157534444-be138d48-a2da-42e8-af2b-560f9e19c4cf.png)

Reply:
that works for me.. though I wonder about accessibility here.
with the icons on the side, even color blind users (of any type) would be able to tell the difference since the symbols were there. I guess hovering to get the status will always stay and can be used in that case

Reply:
> that works for me.. though I wonder about accessibility here. with the icons on the side, even color blind users (of any type) would be able to tell the difference since the symbols were there. I guess hovering to get the status will always stay and can be used in that case

The symbols themselves were always quite small. Arguably, using text-color is better for more people.
I'll upload the changes in a few minutes.